datafusion/common/src/stats.rs

```diff
@@ -151,6 +151,15 @@ impl Precision<usize> {
             (_, _) => Precision::Absent,
         }
     }
+
+    /// Return the estimate of applying a filter with estimated selectivity
+    /// `selectivity` to this Precision. A selectivity of `1.0` means that all
+    /// rows are selected. A selectivity of `0.5` means half the rows are
+    /// selected. Will always return inexact statistics.
+    pub fn apply_filter(self, selectivity: f64) -> Self {
+        self.map(|v| ((v as f64 * selectivity).ceil()) as usize)
```

Comment:
Why use ceil in this case rather than `round`?

Reply:
Maybe Inexact(0) triggers some decisions at somewhere?

Reply:
Yeah, I think there were several cases where DataFusion was (incorrectly) optimizing away scans based on `Inexact(0)` -- I tried to capture some of what is going on in the  https://github.com/apache/arrow-datafusion/issues/8227 ticket if you are interested

Reply:
> Why use ceil in this case rather than round?

Also, specifically in this PR I used `ceil()` because that is what the existing code did in the older codepath.